clippy_lints/src/methods/mod.rs

```diff
@@ -2908,18 +2932,85 @@ fn lint_skip_while_next<'tcx>(
     }
 }
 
-/// lint use of `filter().map()` for `Iterators`
+fn is_method<'tcx>(cx: &LateContext<'tcx>, expr: &hir::Expr<'_>, method_path: &[&str], method_name: Symbol) -> bool {
+    match &expr.kind {
+        hir::ExprKind::Path(qp) => match_qpath(qp, method_path),
+        hir::ExprKind::Closure(_, _, c, _, _) => {
+            let body = cx.tcx.hir().body(*c);
+            let arg_id = body.params[0].pat.hir_id;
+            match body.value.kind {
+                hir::ExprKind::MethodCall(hir::PathSegment { ident, .. }, _, ref args, _) => {
+                    if_chain! {
+                    if ident.name == method_name;
+                    if let hir::ExprKind::Path(path) = &args[0].kind;
+                    if let Res::Local(ref local) = cx.qpath_res(path, args[0].hir_id);
+                    then {
+                        return arg_id == *local
+                    }
+                    }
+                    false
+                },
+                hir::ExprKind::Block(ref block, _) => block
+                    .expr
+                    .as_ref()
+                    .map_or(false, |expr| is_method(cx, &expr, method_path, method_name)),
+                _ => false,
+            }
+        },
+        _ => false,
+    }
+}
+
+// these two paths can't be checked and thus aren't in `paths`
+const OPTION_IS_SOME: [&str; 4] = ["core", "option", "Option", "is_some"];
```

Comment:
I added these just in time to collide with the path-checking dogfooding we added. They're AFAIK not checkable as paths, so I put them here. Thoughts? Top of this file?

Reply:
Why aren't those not checkable by the internal lint? Either this is a bug in the internal lint or the paths are just wrong.

Reply:
Ah! I fixed this in #6567.